source/Octopus.Tentacle.Tests.Integration/CapabilitiesServiceV2Test.cs

```diff
@@ -32,17 +32,38 @@ public async Task CapabilitiesFromAnOlderTentacleWhichHasNoCapabilitiesService_W
 
             //all versions have ScriptServiceV1 & IFileTransferService
             var expectedCapabilitiesCount = 2;
-            if (version.HasScriptServiceV3Alpha())
+            if (version.HasScriptServiceV2())
             {
-                capabilities.Should().Contain("IScriptServiceV3Alpha");
+                capabilities.Should().Contain("IScriptServiceV2");
                 expectedCapabilitiesCount++;
             }
+
+            capabilities.Count.Should().Be(expectedCapabilitiesCount);
+        }
+
+        [Test]
+        [TentacleConfigurations]
+        public async Task CapabilitiesServiceDoesNotReturnKubernetesScriptServiceForNonKubernetesTentacle(TentacleConfigurationTestCase tentacleConfigurationTestCase)
+        {
+            var version = tentacleConfigurationTestCase.Version;
+
+            await using var clientAndTentacle = await tentacleConfigurationTestCase.CreateLegacyBuilder().Build(CancellationToken);
+
+            var capabilities = (await clientAndTentacle.TentacleClient.CapabilitiesServiceV2.GetCapabilitiesAsync(new(CancellationToken))).SupportedCapabilities;
+
+            capabilities.Should().Contain("IScriptService");
+            capabilities.Should().Contain("IFileTransferService");
+
+            //all versions have ScriptServiceV1 & IFileTransferService
+            var expectedCapabilitiesCount = 2;
             if (version.HasScriptServiceV2())
             {
                 capabilities.Should().Contain("IScriptServiceV2");
                 expectedCapabilitiesCount++;
             }
 
+            capabilities.Should().NotContain("IKubernetesServiceV1Alpha");
```

Comment:
Should we make this more general? match on anything with `Kubernetes` in it?

Reply:
I think having it specific makes sense for now. If we get another Kubernetes service, we can re-evaluate it